### ISSUE TITLE
Add a hook 'actionValidateOrderAfter', This hook is called after the complete creation of an order

### ISSUE THREAD

**Issue by olecorre** [PrestaShop/PrestaShop#23787]
#### Is your feature request related to a problem?

When an order is created, several modules can be called by the validateOrder hook or other hooks.
There are very often problems and an order is not completely created, no status, no confirmation email sent etc... the reasons can be multiple, poorly developed module, unexpected error, call to an api that does not respond.
Linking PrestaShop to an ERP is more and more common and the exchanges must be reliable.
Also sending to an ERP can generate a change of status for the order but since it is not completely created, there can be conflicts.

#### Describe the solution you'd like

This hook allows to create the order completely and then the hook is executed, sending the order to an ERP for example, allows to have the whole order with a status etc. and if the link with the ERP does not work, the order is not "corrupted".
I took the example of link with an ERP because I added this hook on several sites and it is very useful.

#### Alternatives you've considered

the validateOrder hook is very often used but causes a lot of maintenance

#### Additional context

create a module that grafts to this hook, make it so that the hook in the module does not respond.
The command will still be created correctly, unlike ValidateOrder

**Comment by florine2623** [comment]
Hello @olecorre ,

Thanks for the suggestion and PR.

Can you follow the issue report, so we can test the issue properly :
![Capture d’écran 2021-03-26 à 15 42 04](https://user-images.githubusercontent.com/16019289/112648438-d3141e80-8e49-11eb-8672-60eec7e07283.png)

Thanks!

**Comment by hibatallahAouadni** [comment]
Hello @olecorre 

Thanks for the suggestion, the Developer Team will take a look.
Please stay tuned !

Ping @PrestaShop/prestashop-core-developers, do you think adding this hook **actionValidateOrderAfter** is a must have?

**Comment by jf-viguier** [comment]
I suggest to add the same arguments as actionValidateOrder :
```
'cart' => $this->context->cart,
'order' => $order,
'customer' => $this->context->customer,
'currency' => $this->context->currency,
'orderStatus' => $order_status
```
The cart, orderStatus can be usefull

**Comment by jf-viguier** [comment]
If someone hooked on actionValidateOrder change to actionValidateOrderAfter, he can expert the same parameters.

**Comment by olecorre** [comment]
I have adding the same arguments as actionValidateOrder

**Comment by Progi1984** [comment]
Fixed in #24738 by @olecorre
